### PR TITLE
v2.0.0

### PR DESCRIPTION
**Pull request by johnnycoinbase**
- sample project
- v2 transaction, transfer, and account 
- retrofit-style callback interface for v2 methods
- jcenter and gradle integration

### CODE REVIEW

File: coinbase-java/src/main/java/com/coinbase/CoinbaseImpl.java [hunk 1]
```diff
@@ -1344,4 +1440,329 @@ private Request newAccountSpecificRequest() {
...
+                        .addHeader("CB-ACCESS-TIMESTAMP", timestamp)
+                        .build();
+
+//                com.squareup.okhttp.Response response = chain.proceed(newRequest);
+//
+//                int tryCount = 0;
+//                while (!response.isSuccessful() && tryCount < 3) {
+//                    tryCount++;
+//                    response = chain.proceed(newRequest);
+//                }
```
Comment (janechung) [inline coinbase-java/src/main/java/com/coinbase/CoinbaseImpl.java]: Should we remove these comments?

File: app/src/main/java/com/coinbase/coinbasesample/MainActivity.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            errors = converter.convert(response.errorBody());
+            if (errors.getErrors().size() > 0)
+                message = errors.getErrors().get(0).getMessage();
+        } catch (IOException e) {
+
+        }
+
+        return message;
+    }
+}
```
Comment (janechung) [inline app/src/main/java/com/coinbase/coinbasesample/MainActivity.java]: This is only my opinion, but when I download sample projects, those that have well documented comments are always easier to follow. Could we potentially add some comments?

Comment (johnnycoinbase) [inline app/src/main/java/com/coinbase/coinbasesample/MainActivity.java]: Yeah, the plan is definitely to flesh out the sample project more. The next PR will include more/better comments among other things.

File: coinbase-java/src/main/java/com/coinbase/CoinbaseImpl.java (hunk 1 above)
Comment (irufus) [inline coinbase-java/src/main/java/com/coinbase/CoinbaseImpl.java]: Code smell ewwwwwwwwwwwww

**Comment by janechung** [comment]
:+1:

**Pull request merged**